test/test_common/logging.h

```diff
@@ -61,20 +61,38 @@ class LogRecordingSink : public Logger::SinkDelegate {
   std::vector<std::string> messages_;
 };
 
+typedef std::vector<std::pair<std::string, std::string>> ExpectedLogSequence;
+
+// Validates that when the stmt is executed, a sequence of emmitted log messages matches the
+// sequence of expected logs/log level pair (ExpectedLogSequence). Note that both sequences must
+// respect the same order.
+#define EXPECT_LOG_SEQ(expected_log_sequence, stmt)                                                \
+  do {                                                                                             \
+    LogLevelSetter save_levels(spdlog::level::trace);                                              \
+    LogRecordingSink log_recorder(Logger::Registry::getSink());                                    \
+    stmt;                                                                                          \
+    ASSERT_EQ(expected_log_sequence.size(), log_recorder.messages().size());                       \
```

Comment:
As I said before, if a new message gets added, the test ends here without actionable info.  I think all test failures should show expected/actual.

My recommendation is to join the actual logs with "\n" and compare that to the expected, which can also be joined.  Then you can use EXPECT_EQ.

Reply:
@jmarantz I thought about doing that, but then we would need to assume that the expected and the actual logs are exactly the same. My last change shows expected/actual but still allows to do substring/log level lookups. Please, lemme know if this is sufficient. Thanks!